src/calculator/index.ts

```diff
@@ -266,10 +277,138 @@ export default class Calculator {
     return augmented;
   }
 
+  /**
+   * Estimates matching for a given project and potential additional votes
+   * @param potentialVotes
+   * @param roundId
+   */
+  async estimateMatching(
+    potentialVotes: PotentialVote[],
+    roundId: string
+  ): Promise<MatchingEstimateResult[]> {
+    const votes = await this.parseJSONFile<Vote>(
+      "votes",
+      `${this.chainId}/rounds/${this.roundId}/votes.json`
+    );
+
+    const rounds = await this.parseJSONFile<Round>(
+      "rounds",
+      `${this.chainId}/rounds.json`
+    );
+
+    const round = rounds.find((r: Round) => r.id === this.roundId);
+
+    if (round === undefined) {
+      throw new ResourceNotFoundError("round");
+    }
+
+    if (round.matchAmount === undefined) {
+      throw new ResourceNotFoundError("round match amount");
+    }
+
+    if (round.token === undefined) {
+      throw new ResourceNotFoundError("round token");
+    }
+
+    const applications = await this.parseJSONFile<Application>(
+      "applications",
+      `${this.chainId}/rounds/${this.roundId}/applications.json`
+    );
+
+    const currentResults = await this._calculate(votes);
+    const potentialVotesAugmented: Vote[] = await Promise.all(
+      potentialVotes.map(async (vote) => {
+        const { amount: amountUSD } = await this.priceProvider.convertToUSD(
+          this.chainId,
+          vote.token,
+          vote.amount
+        );
+
+        const { amount: amountRoundToken } =
+          await this.priceProvider.convertFromUSD(
+            this.chainId,
+            round.token,
+            amountUSD
+          );
+
+        /* Find the latest approved application */
+        const application = applications
+          .filter(
+            (application) =>
+              application.metadata?.application.recipient === vote.recipient
+          )
+          .filter((application) => application.status === "APPROVED")
```

Comment:
We can skip the check on the application status because we are not indexing but just simulating the results on demand for the current request.
If a client sends an application not approved it will get back wrong results, bad for itself but it's not saved anywhere, and since we are building the request in Grants Stack, we know the response will be good for us.